sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -168,6 +166,24 @@ pi_result OCL(piPlatformsGet)(pi_uint32 num_entries, pi_platform *platforms,
   return static_cast<pi_result>(result);
 }
 
+pi_result OCL(piextDeviceInterop)(pi_device *device, void **handle) {
+  // The PI device is the same as OpenCL device handle.
+  assert(device);
+  assert(handle);
+
+  if (*device == nullptr) {
+    // unitialized *device.
+    assert(*handle);
+    *device = cast<pi_device>(*handle);
+  } else {
+    assert(*handle == nullptr);
+    *handle = *device;
+  }
+
+  clRetainDevice(cast<cl_device_id>(*handle));
```

Comment:
1. Why do you retain device here?
2. please, check return code of clRetainDevice.

Reply:
The piextDeviceInterop carries the "retain" semantics by design (not to create
"dangling" PI objects), and note that the "retain" is gone from where the piextDeviceInterop is called.

With that maybe a better name would be piextDeviceRetainInterop

Reply:
> and note that the "retain" is gone from where the piextDeviceInterop is called.
> 
I am still having a piDeviceRetian for the constructors called using a piDevice handle. 
I am now guarding it with the interoperability bool. So if we delete the call to clRetainDevice, we can keep the piDeviceRetain() call.

Reply:
Not doing the retain in piextDeviceInterop is wrong in my opinion because it would mean creating a PI device object that is not used (ref-count 0), which might present nasty problems if not immediately followed by piDeviceRetain.

Reply:
@romanovvlad  Can you share your concerns regarding this?

Reply:
I think this is resolved.